Trimming fails when used on a classic CSproj

Make a console app targeting net461.

Add a reference to Microsoft.Packaging.Tools.Trimming.

Observe:
```
F:\testNetfx\classicProj\packages\Microsoft.Packaging.Tools.Trimming.1.0.0-preview2-25315-0\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4044: The "TrimFiles" task was not given a value for the required parameter "TargetFramework". [F:\testNetfx\classicProj\classicProj.csproj]
```

To workaround set 
```
<NuGetTargetMoniker>.NETFramework,Version=v4.6.1</NuGetTargetMoniker>
```

After doing so the task works correctly.

Originally the task was written such that it didn't attempt to handle the packages.config case, thus this issue.

Given the usefulness of trimming in light of the 100 assemblies added for OOB ns2.0 on desktop I think we should enable this.  I think it'd be OK to enable it without package rooting / package graph (which we don't have for packages.config).  It would just mean that if folks wanted to root more they'd have to root the files.

/cc @Petermarcu

--- a/Microsoft.Packaging.Tools.Trimming/tasks/FileNode.cs
+++ b/Microsoft.Packaging.Tools.Trimming/tasks/FileNode.cs
@@ -58,6 +58,11 @@ namespace Microsoft.DotNet.Build.Tasks
         public NuGetPackageNode Package { get; }
         public IEnumerable<FileNode> Dependencies { get { return dependencies; } }
 
+        public override string ToString()
+        {
+            return Name;
+        }
+
         public void PopulateDependencies(IDictionary<string, FileNode> allFiles, bool preferNativeImage, ILog log)
         {
             if (dependencies == null)
@@ -73,15 +78,17 @@ namespace Microsoft.DotNet.Build.Tasks
                 stack = new Stack<FileNode>();
             }
 
-            stack.Push(this);
+            if (stack.Contains(this))
+            {
+                log.LogMessage($"Cycle detected: {String.Join(" -> ", stack)} -> {this}");
+            }
 
             if (dependencies != null)
             {
-                // re-entrant call indicates a cycle, bail
-                log.LogMessage($"Cycle detected: {String.Join(" -> ", stack)}");
-                stack.Pop();
                 return;
             }
+
+            stack.Push(this);
 
             dependencies = new HashSet<FileNode>();
 

--- a/Microsoft.Packaging.Tools.Trimming/tasks/TrimFiles.cs
+++ b/Microsoft.Packaging.Tools.Trimming/tasks/TrimFiles.cs
@@ -50,7 +50,6 @@ namespace Microsoft.DotNet.Build.Tasks
         /// <summary>
         /// Target framework to use when determining package dependencies.  Should be long form, IE: .NETCoreApp,Version=v1.0
         /// </summary>
-        [Required]
         public string TargetFramework { get; set; }
 
         /// <summary>
@@ -250,6 +249,12 @@ namespace Microsoft.DotNet.Build.Tasks
         internal IDictionary<string, NuGetPackageNode> GetPackagesFromPackageDependencies()
         {
             Dictionary<string, NuGetPackageNode> packages = new Dictionary<string, NuGetPackageNode>(StringComparer.OrdinalIgnoreCase);
+
+            if (string.IsNullOrEmpty(TargetFramework) && PackageDependencies?.Length > 0)
+            {
+                Log.LogError($"{nameof(TargetFramework)} was not specified and is required when {nameof(PackageDependencies)} are present.");
+                return packages;
+            }
 
             var target = TargetFramework;
 

